### PR TITLE
Cleaned up generators tests using internal assertion helper

### PR DESCRIPTION
**Pull request by ronakjangir47**
/cc @kaspth

### CODE REVIEW

File: railties/test/generators_test.rb [hunk 1]
```diff
@@ -51,23 +47,20 @@ def test_help_when_a_generator_with_required_arguments_is_invoked_without_argume
 
   def test_should_give_higher_preference_to_rails_generators
     assert File.exist?(File.join(@path, 'generators', 'model_generator.rb'))
-    @mock_generator.expect(:call, nil, [["Account"],{}])
-    Rails::Generators::ModelGenerator.stub(:start, @mock_generator) do
+    assert_called_with(Rails::Generators::ModelGenerator, :start, [["Account"],{}]) do
```
Comment (rafaelfranca) [inline railties/test/generators_test.rb]: Missing space before `{`

File: railties/test/generators_test.rb [hunk 2]
```diff
@@ -51,23 +47,20 @@ def test_help_when_a_generator_with_required_arguments_is_invoked_without_argume
...
       warnings = capture(:stderr){ Rails::Generators.invoke :model, ["Account"] }
       assert warnings.empty?
     end
   end
 
   def test_invoke_with_default_values
-    @mock_generator.expect(:call, nil, [["Account"],{}])
-    Rails::Generators::ModelGenerator.stub(:start, @mock_generator) do
+    assert_called_with(Rails::Generators::ModelGenerator, :start, [["Account"],{}]) do
```
Comment (rafaelfranca) [inline railties/test/generators_test.rb]: same here

**Comment by ronakjangir47** [comment]
@rafaelfranca Updated PR

File: railties/test/generators/actions_test.rb [hunk 3]
```diff
@@ -150,16 +147,13 @@ def test_environment_with_block_should_include_block_contents_in_environment_ini
...
+    assert_called_with(generator, :run, ['git init']) do
       action :git, :init
     end
   end
 
   def test_git_with_hash_should_run_each_command_using_git_scm
-    @mock_generator.expect(:call, nil, ["git rm README"])
-    @mock_generator.expect(:call, nil, ["git add ."])
-    generator.stub(:run, @mock_generator) do
+    assert_called(generator, :run, times: 2) do
```
Comment (senny) [inline railties/test/generators/actions_test.rb]: In my opinion this kind of test refactoring open potential holes in our suite. The assertions used to be specific, now with `times: 2` they are very generalized and don't have much meaning left.

Comment (ronakjangir47) [inline railties/test/generators/actions_test.rb]: @senny previously this test was same as now. please see #20357

````diff
   def test_git_with_hash_should_run_each_command_using_git_scm
-    generator.expects(:run).times(2)
-    action :git, rm: 'README', add: '.'
+    @mock_generator.expect(:call, nil, ["git rm README"])
+    @mock_generator.expect(:call, nil, ["git add ."])
+    generator.stub(:run, @mock_generator) do
+      action :git, rm: 'README', add: '.'
+    end
   end
````
At that time we did not have these assertion helpers so I did it this way.

Comment (senny) [inline railties/test/generators/actions_test.rb]: thanks for clarifying. I'll leave it up to @kaspth or @rafaelfranca.

Comment (kaspth) [inline railties/test/generators/actions_test.rb]: Can't we do it like this?

```ruby
assert_called_with(generator, :run, [ ["git rm README"], ["git add ."] ]) do
  action :git, rm: 'README', add: '.'
end
```

File: railties/test/generators/generators_test_helper.rb [hunk 4]
```diff
@@ -1,8 +1,10 @@
 require 'abstract_unit'
 require 'active_support/core_ext/module/remove_method'
 require 'active_support/testing/stream'
+require 'active_support/testing/method_call_assertions'
 require 'rails/generators'
 require 'rails/generators/test_case'
+require 'minitest/mock'
```
Comment (kaspth) [inline railties/test/generators/generators_test_helper.rb]: Let move that require to `method_call_assertions`.

**Comment by ronakjangir47** [comment]
@kaspth done

**Comment by kaspth** [comment]
Thanks

**Comment by ronakjangir47** [comment]
Thanks @kaspth

**Pull request merged**